### PR TITLE
Removing Moq hard-dependency on AddProbe method

### PR DESCRIPTION
**Pull request by HuwLittle**
Removing Moq hard-dependency on AddProbe method

Adding tests

### CODE REVIEW

**Comment by HuwLittle** [comment]
@seniorquico - does this look ok to you?

**Comment by seniorquico** [comment]
Sorry, @HuwLittle. I'm out sick. It might be a little bit before I have time to review this. Thanks for your understanding.

**Comment by HuwLittle** [comment]
no problem at all, hope you get well soon

**Comment by HuwLittle** [comment]
Hey @seniorquico - I hope you're feeling a little better, is there another maintainer of this library that may be able to take a look in your absence?

Thank you.

**Comment by HuwLittle** [comment]
Just a poke on this @seniorquico - hope you're feeling better.

**Comment by HuwLittle** [comment]
Hi @seniorquico - is there someone else I can go to to get this looked at. These PRs are quite important for the orleans work I'm doing at my job, so would really appreciate them being looked at as soon as possible.

**Comment by seniorquico** [comment]
@HuwLittle I should be able to get to this over the weekend. 👍

File: src/OrleansTestKit/TestGrainFactory.cs [hunk 1]
```diff
@@ -100,8 +99,14 @@ internal Mock<T> AddProbe<T>(IGrainIdentity identity, string classPrefix = null)
             return mock;
         }
 
+        internal void AddProbe<T>(Func<IGrainIdentity, T> factory) where T : class, IGrain => _probeFactories.Add(typeof(T), factory);
+
         internal void AddProbe<T>(Func<IGrainIdentity, IMock<T>> factory)
-            where T : class, IGrain => _probeFactories.Add(typeof(T), factory);
+            where T : class, IGrain
+        {
+            var adaptedFactory = new Func<IGrainIdentity, T>(grainIdentity => factory(grainIdentity).Object);
```
Comment (seniorquico) [inline src/OrleansTestKit/TestGrainFactory.cs:107]: The (removed) line 139 below would've tolerated a user-provided function that returned `null`.

```cs
grain = mock?.Object;
```

A user shouldn't do this, but to preserve existing behaviors we should pass through the `null` value.

```suggestion
            var adaptedFactory = new Func<IGrainIdentity, T>(grainIdentity => factory(grainIdentity)?.Object);
```

File: test/OrleansTestKit.Tests/Tests/GrainProbeTests.cs [hunk 2]
```diff
@@ -143,8 +146,9 @@ public async Task FactoryProbe()
             pong.Verify(p => p.Pong(), Times.Once);
         }
 
+        #region FactoryProbeWithMultipleNewProbes
```
Comment (seniorquico) [inline test/OrleansTestKit.Tests/Tests/GrainProbeTests.cs:149]: Please remove the regions. This isn't aligned with the current code base's style.

As an aside... I should get `dotnet-format` added to the GitHub Actions pipeline to help flag & address these small things more quickly.

**Review by seniorquico — CHANGES_REQUESTED** [review]

File: test/OrleansTestKit.Tests/Tests/GrainProbeTests.cs (hunk 2 above)
Comment (HuwLittle) [reply test/OrleansTestKit.Tests/Tests/GrainProbeTests.cs:149]: of course, will remove

Comment (HuwLittle) [reply test/OrleansTestKit.Tests/Tests/GrainProbeTests.cs]: done

File: src/OrleansTestKit/TestGrainFactory.cs (hunk 1 above)
Comment (HuwLittle) [reply src/OrleansTestKit/TestGrainFactory.cs]: of course, this is doe

**Review by seniorquico — APPROVED** [review]

**Pull request merged**